pkg/reduce/reduce_test.go

```diff
@@ -672,6 +731,7 @@ func publishMessages(ctx context.Context, startTime int, messages []int, testDur
 					}
 					count = 0
 				}
+				time.Sleep(interval)
```

Comment:
why do we have to sleep?

Reply:
`// publish messages with 1 millisecond delay
	go publishMessages(ctx, startTime, messages, 600, 10, p[fromBuffer.GetName()], fromBuffer, 1*time.Millisecond)
	// sleep for 100 milliseconds, so that few messages are processed and partitions are created
	time.Sleep(100 * time.Millisecond)
	cancel()`

since I am closing the context after 100 milliseconds, I have added sleep just to ensure context close happens while we are processing the data

Reply:
this sleep is only for context close tests, for other tests the sleep duration is 0

Reply:
won't think become flaky?

Reply:
removed sleep